pandas/io/parsers.py

```diff
@@ -1660,15 +1660,17 @@ def _convert_to_ndarrays(self, dct, na_values, na_fvalues, verbose=False,
                     try_num_bool=False)
             else:
                 # skip inference if specified dtype is object
-                try_num_bool = not (cast_type and is_string_dtype(cast_type))
+                try_num_bool = not (cast_type and (is_string_dtype(cast_type)
```

Comment:
don't use a noqa, rather make another variable this condition is hard to read / understand